### ISSUE TITLE
Websocket size constraints

### ISSUE THREAD

**Issue by greg-szabo** [informalsystems/ibc-rs#1299]
## Crate
hermes

## Summary of Bug
I need to understand websocket size constraints better. What does this error message mean: (during Osmosis reward distribution)
```
Aug 17 13:17:59 hermes hermes[7621]: Aug 17 13:17:59.046 ERROR [osmosis-1] failed to collect events: WebSocket driver failed: WebSocket Error: failed to read from WebSocket connection: Space limit exceeded: Message length too big: 202887188 > 16777216 (code: 1)
```

## Version
0.6.1

## Acceptance Criteria
no errors :)

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by adizere** [comment]
@mircea-c wanted to check with you if the problem still appears?

There was a [fix](https://github.com/informalsystems/tendermint-rs/issues/974) from the tm-rs side, which went into 0.22, and Hermes 0.7.2 should have that fix. It's not clear to me, however, if that fixed the problem reported above. Does Hermes still report "Space limit exceeded" ?

**Comment by romac** [comment]
This was actually not released as part of 0.7.2 but only part of a branch and special build of Hermes. I don't know if @mircea-c got a chance to deploy it temporarily and whether it actually fixed the problem?

**Comment by mircea-c** [comment]
Problem still there on the latest osmosis version
```
 2021-12-14T17:23:45.755660Z ERROR ThreadId(61) [osmosis-1] failed to collect events: WebSocket driver failed: web socket error: failed to read from WebSocket connection: Space limit exceeded: Message length too big: 70286072 > 16777216
 2021-12-14T17:23:45.761674Z  INFO ThreadId(61) [osmosis-1] successfully reconnected to WebSocket endpoint ws://10.10.51.208:26690/websocket
```

**Comment by adizere** [comment]
The latest status on websocket improvements is:

- Websocket improvements in tendermint-go (PR [#7230](https://github.com/tendermint/tendermint/issues/7230)) landed into tendermint `0.34.15`
- There is no SDK release to build on tendermint-go `0.34.15`:
  - SDK 0.45 line latest release is 0.45.1 and is [using tendermint `0.34.14`](https://github.com/cosmos/cosmos-sdk/blob/2646b474c7beb0c93d4fafd395ef345f41afc251/go.mod#L49)
  - SDK 0.46-alpha2 released, but [uses tendermint `0.35`](https://github.com/cosmos/cosmos-sdk/blob/428ae3c15c1729197291e2a3fe77013f255cefb5/go.mod#L50)
  - So effectively the changes from #7230 did not land anywhere.

This is a bit of a disappointment. But since operators did not re-mention problems regarding websocket I'll optimistically close this.

**Comment by JakeHartnell** [comment]
> The latest status on websocket improvements is:
> 
> * Websocket improvements in tendermint-go (PR [#7230](https://github.com/tendermint/tendermint/issues/7230)) landed into tendermint `0.34.15`
> * There is no SDK release to build on tendermint-go `0.34.15`:
>   
>   * SDK 0.45 line latest release is 0.45.1 and is [using tendermint `0.34.14`](https://github.com/cosmos/cosmos-sdk/blob/2646b474c7beb0c93d4fafd395ef345f41afc251/go.mod#L49)
>   * SDK 0.46-alpha2 released, but [uses tendermint `0.35`](https://github.com/cosmos/cosmos-sdk/blob/428ae3c15c1729197291e2a3fe77013f255cefb5/go.mod#L50)
>   * So effectively the changes from #7230 did not land anywhere.
> 
> This is a bit of a disappointment. But since operators did not re-mention problems regarding websocket I'll optimistically close this.

Building some really cool IBC applications that need this. Just came across this issue. Could we reopen? :joy:

**Comment by romac** [comment]
Do you mean you are running  into problems with the WebSocket connection? If so, then yes please re-open the issue and attach a description of the issues you are facing, together with logs if possible. Thanks :)
